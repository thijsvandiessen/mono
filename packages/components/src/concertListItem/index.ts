export { ConcertListItem, type Props } from './concertListItem.jsx'
