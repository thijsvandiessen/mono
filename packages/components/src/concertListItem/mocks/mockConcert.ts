import type { Concert } from '@mono/data'

export const mockConcert: Concert = {
  id: 'id',
  title: 'name',
  image: undefined,
  locations: [
    {
      startTime: '2023-09-01T13:32:36+01:00',
      id: 'test id',
      title: 'test',
      address: 'test',
      ticketLink: 'https://example.com',
      lat: 0,
      lng: 0,
    },
  ],
  url: 'some-slug',
  content: [],
}
