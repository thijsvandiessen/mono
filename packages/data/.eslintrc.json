{
  "ignorePatterns": ["**/generated/*.ts", "dist/*"],
  "plugins": ["@typescript-eslint", "testing-library"],
  "extends": [
    "eslint:recommended",
    "plugin:react-hooks/recommended",
    "plugin:@typescript-eslint/recommended"
  ],
  "parser": "@typescript-eslint/parser",
  "env": {
    "jest": true
  },
  "rules": {
    "require-await": ["warn"],
    "no-unused-vars": ["warn"],
    "@typescript-eslint/no-explicit-any": "warn",
    "@typescript-eslint/consistent-type-imports": "error",
    "sort-imports": [
      "warn",
      {
        "ignoreCase": false,
        "ignoreDeclarationSort": false,
        "ignoreMemberSort": false,
        "memberSyntaxSortOrder": ["none", "all", "multiple", "single"],
        "allowSeparatedGroups": false
      }
    ]
  },
  "root": true
}
