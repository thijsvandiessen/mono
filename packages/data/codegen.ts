import type { CodegenConfig } from '@graphql-codegen/cli'

const config: CodegenConfig = {
  schema: './src/generated/schema.graphql',
  documents: ['./**/*.ts'],
  emitLegacyCommonJSImports: false,
  generates: {
    './src/generated/': {
      preset: 'client',
      config: {
        withHooks: true,
        scalars: {
          BooleanType: 'boolean',
          CustomData: 'Record<string, string>',
          Date: 'string',
          DateTime: 'string',
          FloatType: 'number',
          IntType: 'number',
          ItemId: 'string',
          JsonField: 'unknown',
          MetaTagAttributes: 'Record<string, string>',
          UploadId: 'string',
        },
        useTypeImports: true,
        verbatimModuleSyntax: true,
      },
      presetConfig: {
        persistedDocuments: true,
        gqlTagName: 'gql',
        fragmentMasking: false,
      },
    },
    './src/generated/introspection.json': {
      plugins: ['introspection'],
      config: {
        minify: false,
        schemaDescription: true,
        federation: true,
      },
    },
  },
  hooks: { afterAllFileWrite: ['prettier --write'] },
  ignoreNoDocuments: true,
}

export default config
