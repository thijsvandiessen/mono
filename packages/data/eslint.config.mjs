import { defineConfig, globalIgnores } from 'eslint/config'
import { eslintConfig } from '@mono/eslint-config'

export default defineConfig([
  globalIgnores(['src/generated/*.ts', 'dist/*', 'node_modules']),
  eslintConfig,
])
