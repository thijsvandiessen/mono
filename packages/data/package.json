{
  "name": "@mono/data",
  "version": "1.0.0",
  "description": "data",
  "type": "module",
  "exports": {
    ".": {
      "import": {
        "types": "./dist/index.d.ts",
        "default": "./dist/index.js"
      }
    }
  },
  "scripts": {
    "build": "tsc",
    "codegen": "graphql-codegen-esm --config ./codegen.ts",
    "codegen:watch": "graphql-codegen-esm --require dotenv/config --config ./codegen.ts --watch",
    "dev": "tsc --watch",
    "download-schema": "rover graph fetch vandiessen-mono-graph@current > ./src/generated/schema.graphql && pnpm run lint:styles",
    "lint": "pnpm run lint:js && pnpm run lint:styles",
    "lint:js": "eslint",
    "lint:styles": "prettier . --write  --list-different",
    "test": "vitest run -c ./vitest.config.ts"
  },
  "dependencies": {
    "@graphql-typed-document-node/core": "3.2.0",
    "@mono/next-js": "workspace:*",
    "@mono/utils": "workspace:*",
    "@urql/core": "6.0.1",
    "@urql/exchange-graphcache": "8.1.0",
    "@urql/exchange-retry": "2.0.0",
    "graphql": "16.11.0",
    "react": "19.2.0"
  },
  "devDependencies": {
    "@apollo/rover": "0.36.2",
    "@commitlint/config-conventional": "20.0.0",
    "@graphql-codegen/cli": "6.0.1",
    "@graphql-codegen/client-preset": "5.1.1",
    "@graphql-codegen/introspection": "5.0.0",
    "@graphql-codegen/typescript": "5.0.2",
    "@graphql-codegen/typescript-document-nodes": "5.0.2",
    "@graphql-codegen/typescript-operations": "5.0.2",
    "@mono/eslint-config": "workspace:*",
    "@mono/typescript-config": "workspace:*",
    "@testing-library/jest-dom": "6.9.1",
    "@testing-library/react": "16.3.0",
    "@types/node": "24.9.2",
    "@types/react": "19.2.2",
    "@types/react-dom": "19.2.2",
    "@vitejs/plugin-react": "5.1.0",
    "dotenv": "17.2.3",
    "eslint": "9.38.0",
    "eslint-plugin-testing-library": "7.13.3",
    "prettier": "3.6.2",
    "turbo": "2.5.8",
    "typescript": "5.9.3",
    "vite-tsconfig-paths": "5.1.4",
    "vitest": "4.0.4"
  }
}
