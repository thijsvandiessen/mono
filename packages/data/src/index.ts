export { client } from './gqlClient.js'
export {
  getEvent,
  getEventPage,
  getEventPageSeo,
  getEvents,
  getEventsMeta,
  getEventUrls,
  getFutureEvents,
  getGeneralInfo,
  getLocation,
  getLocations,
  getPage,
  getPages,
  getPageSeo,
  getPagesMeta,
  getSiteInfo,
  getSiteMetadata,
} from './getters/index.js'

export type {
  Author,
  BlogPost,
  Event,
  PageLink,
  CloudinaryAsset,
  Image,
  Location,
  Music,
} from './types/index.js'

export { isOfTypeCloudinaryAsset } from './types/index.js'
export {
  formatCloudinaryDocument,
  formatCloudinaryImage,
} from './formatters/index.js'
export * from './generated/graphql.js'
