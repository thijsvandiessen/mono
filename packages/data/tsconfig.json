{
  "extends": "@mono/typescript-config/base.json",
  "compilerOptions": {
    "lib": ["ESNext", "DOM"]
  },
  "include": ["jest.setup.cjs", "src/**/*", "src/*"]
}
