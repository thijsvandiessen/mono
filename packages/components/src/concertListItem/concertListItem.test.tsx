import { describe, expect, it, vi } from 'vitest'
import { render, screen } from '@testing-library/react'
import { ConcertListItem } from './concertListItem.jsx'
import React from 'react'
import { getConcert } from '@mono/data'
import { mockConcert } from './mocks/mockConcert.js'

vi.mock('../location/index.js', () => {
  const originalModule = vi.importActual<typeof import('../location/index.js')>(
    '../location/index.js'
  )
  return {
    __esModule: true,
    ...originalModule,
    Location: vi.fn(() => <p>Location Component</p>),
  }
})

vi.mock('@mono/data', () => {
  const originalModule =
    vi.importActual<typeof import('@mono/data')>('@mono/data')
  return {
    __esModule: true,
    ...originalModule,
    getConcert: vi.fn(),
  }
})

const getConcertMock = vi.mocked(getConcert)

describe('Concert component', () => {
  it('shows all the data', () => {
    getConcertMock.mockResolvedValue({
      data: mockConcert,
      error: undefined,
    })

    const { container } = render(<ConcertListItem data={mockConcert} />)

    expect(container).toMatchSnapshot()

    expect(screen.getByText('name')).toBeInTheDocument()
  })
})
