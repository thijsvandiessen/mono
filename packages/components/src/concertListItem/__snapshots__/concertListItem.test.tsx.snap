// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Concert component > shows all the data 1`] = `
<div>
  <div
    class="_root_2436e2 _oneLocation_2436e2"
  >
    <h2
      class="_title_2436e2 h3"
    >
      <span>
        <a
          href="some-slug"
        >
          name
        </a>
      </span>
    </h2>
    <div
      class="_content_2436e2"
    >
      <ul
        class="_locations_2436e2"
      >
        <li
          class="_location_2436e2"
        >
          <h3
            class="_locationTitle_2436e2 h5"
          >
            test
          </h3>
          <p
            class="_locationSubtitle_2436e2"
          >
            Vrijdag 1 september â€” 14:32
          </p>
          <p
            class="_locationLinkContainer_2436e2 text-small"
          >
            Concert voorbij
          </p>
        </li>
      </ul>
    </div>
    <a
      class="_concertLink_2436e2 text-small"
      href="some-slug"
    >
      Bekijk concert
      <svg
        aria-hidden="true"
        class="_icon_884e08"
        height="24px"
        viewBox="0 0 24 24"
        width="24px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          fill="currentColor"
        >
          <path
            d="M23.22 12.84H.78a.781.781 0 0 1 0-1.56h22.45c.43 0 .78.35.78.78-.01.43-.36.78-.79.78z"
          />
          <path
            d="M17.8 18.27c-.43 0-.78-.35-.78-.78 0-.21.08-.4.23-.55l4.88-4.88-4.88-4.88a.785.785 0 0 1-.03-1.1c.29-.31.79-.32 1.1-.03l.03.03 5.43 5.43c.3.3.3.79 0 1.1l-5.43 5.43a.78.78 0 0 1-.55.23z"
          />
        </g>
      </svg>
    </a>
  </div>
</div>
`;
